src/core/or/or_circuit_st.h

```diff
@@ -61,6 +61,13 @@ struct or_circuit_t {
    *  statistics. */
   unsigned int circuit_carries_hs_traffic_stats : 1;
 
+  /** If set, this circuit got a BEGIN_DIR, meaning a directory request, and
+   * not for any hidden service documents. We use this flag for SENDME
+   * mechanism in order to know if this circuit will serve a consensus for
+   * which we need to obey a specific consensus parameter about the minimal
+   * SENDME version we accept. */
+  unsigned int begin_dir_seen : 1;
```

Comment:
I think this might be backwards: If I understand correctly, we don't want somebody to be able to send a BEGIN_DIR in order to make all a circuit exempt from the minimum.  I think instead we wanted to have it so that if the circuit got a regular BEGIN, we know that it is _not_ a directory circuit?

Reply:
Hmmm interesting, I haven't thought of gaming this by sending a `RELAY_BEGIN_DIR` and then performing other tasks than a directory request...

Actually, is this assertion always true?: if no BEGIN than must be a directory circuit?

In other words, it is safe and reliable to apply this minimal version for the SENDME as long as we haven't seen a `BEGIN` on it?